lib/improver/nbhood.py

```diff
@@ -249,8 +361,18 @@ def run(cube, radius_in_km):
             Cube containing the smoothed field after the square neighbourhood
             method has been applied.
         """
+        if isinstance(cube.data, np.ma.MaskedArray):
```

Comment:
Yeah, this is partly why I asked about this yesterday. I think a check for NaNs would be good. I'm not sure about checking for a fill_value. If it is masked and uses a different fill value then it should be picked up by your masked data check. What do you think?